uitools/toolkitcpp/cpp/Esri/ArcGISRuntime/Toolkit/BarChartPopupMediaItem.cpp

```diff
@@ -0,0 +1,116 @@
+/*******************************************************************************
+ *  Copyright 2012-2025 Esri
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ ******************************************************************************/
+#include "BarChartPopupMediaItem.h"
+
+// Qt headers
+#include <QBarSet>
+#include <QColor>
+
+// Maps SDK headers
+#include <PopupMedia.h>
+#include <PopupMediaItem.h>
+#include <PopupMediaValue.h>
+
+namespace Esri::ArcGISRuntime::Toolkit {
+
+/*!
+  \class Esri::ArcGISRuntime::Toolkit::BarChartPopupMediaItem
+  \inmodule ArcGISRuntimeToolkit
+  \brief Represents the contents of a \c{MediaPopupElement::media} to be used for a BarSeries in QML.
+  */
+
+/*!
+  \brief Constructor
+  \list
+    \li \a popupMedia - The \l {Esri::ArcGISRuntime::PopupMedia} {PopupMedia} used to populate the view.
+    \li \a jsonColorArray - The JSON array containing the color information for each bar set.
+    \li \a parent - The optional parent QObject.
+  \endlist
+ */
+BarChartPopupMediaItem::BarChartPopupMediaItem(PopupMedia* popupMedia, const QJsonArray& jsonColorArray, QObject* parent)
+  : PopupMediaItem{popupMedia, parent},
+    m_jsonArr{jsonColorArray}
+{
+}
+
+/*!
+  \brief Returns the list of \c QBarSet objects. You must take ownership of the returned QBarSets.
+ */
+QList<QBarSet*> BarChartPopupMediaItem::barSets()
+{
+  QList<QBarSet*> barSets;
+  auto mediaValue = popupMediaItem()->value();
+
+  for (int i = 0; i < mediaValue->data().length(); i++)
```

Comment:
Consider storing the length once so you don't need to make a nested chain of calls for each loop iteration.